evalml/pipelines/components/transformers/samplers/undersampler.py

```diff
@@ -179,9 +178,7 @@ def fit_resample(self, X, y):
             # iterate through the classes we need to undersample and remove the number of samples we need to remove
             for key, value in result.items():
                 indices = y.index[y == key].values
-                indices_to_remove = self.random_state.choice(
-                    indices, value, replace=False
-                )
+                indices_to_remove = random_state.choice(indices, value, replace=False)
```

Comment:
We don't initialize the `np.random.RandomState` in the initlaization of this component in order to make it deterministic. Since we now initialize it during `fit`, we don't need to save the `np.randomstate` value, so this was for simplification

Reply:
Good catch @bchen1116